src/nvim/misc1.c

```diff
@@ -698,7 +694,7 @@ open_line (
 
   /* (State == INSERT || State == REPLACE), only when dir == FORWARD */
   if (p_extra != NULL) {
-    *p_extra = saved_char;              /* restore char that NUL replaced */
+    *p_extra = (char_u)saved_char;  // restore char that NUL replaced
```

Comment:
True, but we are saving an element from an `char_u` pointer here, and I thought this would be changed when the pointer type (eventually) gets changed. But let's keep `first_char` `int` then.